src/util/mod.rs

```diff
@@ -241,6 +247,41 @@ pub fn unescape(s: &str) -> Vec<u8> {
     buf
 }
 
+#[cfg(test)]
+mod unescape_tests {
+    use super::*;
+
+    #[test]
+    fn test_no_escapes() {
+        assert_eq!(unescape(r"ab"), b"ab");
+    }
+
+    #[test]
+    fn test_escaped_escape() {
+        assert_eq!(unescape(r"a\\023"), b"a\\023");
+    }
+
+    #[test]
+    fn test_three_digit_octal() {
+        assert_eq!(unescape(r"a\000"), b"a\0");
```

Comment:
Why test it again since the doctest tests it already?

Reply:
wow! I even didn't know it will be run in the doc test! Sorry for my misleading @kbacha I think you can remove the doc test to make it more clear and removes duplicates.